packages/blade/src/_helpers/types.ts

```diff
@@ -83,6 +86,46 @@ type TestID = {
   testID?: string;
 };
 
+/**
+ * Returns the object as it is in web and sets all properties of object to type `never` in react native
+ */
+type MakeObjectWebOnly<T> = Platform.Select<{
+  web: T;
+  native: Record<keyof T, never>;
+}>;
+
+/**
+ * Similar to `Pick` except this returns `never` when value doesn't exist (native `Pick` returns `unknown`).
+ *
+ * You might have to ts-ignore the non-existing type error while using this. This is done so that you can get jsdoc from actual type.
+ *
+ * E.g. This will pick from ViewStyle prop if value exists else returns undefined.
+ *
+ * ```ts
+ * // @ts-expect-error: T passed here may not neccessarily exist. We return `never` type when it doesn't
+ * native: PickIfExist<ViewStyle, T>;
+ * ```
+ */
+type PickIfExist<T, K extends keyof T> = {
+  [P in K]: P extends keyof T ? T[P] : never;
+};
+
+/**
+ * Picks the types based on the platform (web / native).
+ *
+ * E.g.
+ * ```ts
+ * type CSSObject = PickCSSByPlatform<'display'>
+ * // On Web --> This will be all possible web display properties like `block`, `flex`, `inline`, and more.
+ * // On Native --> This will be just `flex` and `none`
+ * ```
+ */
+type PickCSSByPlatform<T extends keyof CSSObject | keyof ViewStyle> = Platform.Select<{
+  web: Pick<CSSObject, T>;
```

Comment:
why not use PickIfExist here as well?

Reply:
Because we don't have react-native specific props in Box so everything is supported in web. But yup changed it to PickIfExist so in case we add any react-native specific prop in future, it will be handled.